### PR TITLE
BUG: Add check for array lengths in from_arrays method (GH13599)

### LINKED ISSUE #13599: MultiIndex.from_arrays() does not check lenghts
#### Code Sample, a copy-pastable example if possible

```
In [2]: pd.MultiIndex.from_arrays([[1,2,3], ['a', 'b']])
Out[2]: 
MultiIndex(levels=[[1, 2, 3], ['a', 'b']],
           labels=[[0, 1, 2], [0, 1]])

In [3]: pd.MultiIndex.from_arrays([[1,2,3], ['a', 'b']]).values
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-65e6fa14e07f> in <module>()
----> 1 pd.MultiIndex.from_arrays([[1,2,3], ['a', 'b']]).values

/home/nobackup/repo/pandas/pandas/indexes/multi.py in values(self)
    552             values.append(taken)
    553 
--> 554         self._tuples = lib.fast_zip(values)
    555         return self._tuples
    556 

/home/nobackup/repo/pandas/pandas/lib.pyx in pandas.lib.fast_zip (pandas/lib.c:11131)()
    587         it = <flatiter> PyArray_IterNew(arr)
    588         if len(arr) != n:
--> 589             raise ValueError('all arrays must be same length')
    590 
    591         for i in range(n):

ValueError: all arrays must be same length

```

#### Expected Output

The first previous command should already raise an error.

#### output of ``pd.show_versions()``
```

In [4]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: a63bd12529ff309d957d714825b1753d0e02b7fa
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 4.5.0-2-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.utf8
LOCALE: it_IT.UTF-8

pandas: 0.18.1+174.ga63bd12
nose: 1.3.7
pip: 1.5.6
setuptools: 18.4
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.16.0
statsmodels: 0.8.0.dev0+111ddc0
xarray: None
IPython: 5.0.0.dev
sphinx: 1.3.1
patsy: 0.3.0-dev
dateutil: 2.2
pytz: 2012c
blosc: None
bottleneck: 1.1.0dev
tables: 3.2.2
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: None
xlrd: 0.9.4
xlwt: 1.1.2
xlsxwriter: 0.7.3
lxml: None
bs4: 4.4.0
html5lib: 0.999
httplib2: 0.9.1
apiclient: 1.5.0
sqlalchemy: 1.0.11
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.38.0
pandas_datareader: 0.2.1

```

### PR DESCRIPTION
**Pull request by sahildua2305**
- [x] closes #13599 
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
tests!

File: pandas/tests/indexes/test_multi.py [hunk 1]
```diff
@@ -632,6 +632,13 @@ def test_from_arrays_index_series_period(self):
 
         tm.assert_index_equal(result, result2)
 
+    def test_from_arrays_different_lengths(self):
+        # GH13599
+        idx1 = [1, 2, 3]
```
Comment (jreback) [inline pandas/tests/indexes/test_multi.py]: add a positive example (e.g. it doesn't raise)

**Comment by jreback** [comment]
pls add a whatsnew note for 0.19.0

File: pandas/tests/indexes/test_multi.py (hunk 1 above)
Comment (sahildua2305) [inline pandas/tests/indexes/test_multi.py]: I am not sure if I can assert explicitly that the ValueError isn't raised.

Should I simply add an example where array lengths are same and assert the response is the MultiIndex object with valid data?

File: doc/source/whatsnew/v0.19.0.txt [hunk 2]
```diff
@@ -550,3 +550,5 @@ Bug Fixes
 - Bug in ``groupby`` with ``as_index=False`` returns all NaN's when grouping on multiple columns including a categorical one (:issue:`13204`)
 
 - Bug where ``pd.read_gbq()`` could throw ``ImportError: No module named discovery`` as a result of a naming conflict with another python package called apiclient  (:issue:`13454`)
+
+-Bug in ``MultiIndex.from_arrays`` doesn't check for arrays lengths (:issue:`13599`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: doesn't -> did not

array lengths -> input array lengths

File: pandas/tests/indexes/test_multi.py (hunk 1 above)
Comment (jreback) [inline pandas/tests/indexes/test_multi.py]: you simply construct an expected MultiIndex and use ``.equals`` that it is equal to a result that uses ``.from_arrays``. I am sure there are other examples to model from.

Comment (sahildua2305) [inline pandas/tests/indexes/test_multi.py]: @jreback is that test really required? The same input array length case is anyway getting checked in all other `from_arrays` related tests.

Comment (jreback) [inline pandas/tests/indexes/test_multi.py]: better yet make sure to check with a 0-len array (check for all 2 cases, idx1 or idx2 is 0-len)

**Comment by jorisvandenbossche** [comment]
Superceded by https://github.com/pydata/pandas/pull/13728

**Pull request closed without merging**